### ISSUE TITLE
review/product/listAjax/id/{{non existent id}/

### ISSUE THREAD

**Issue by ruthgeridema** [magento/magento2#13102]
### Preconditions
1. All magento 2 versions

### Steps to reproduce
1. Visit review/product/listAjax/id/{{non existent id}/

### Expected result
1. I would expect a 404 not found

### Actual result
https://i.gyazo.com/c0c9b0218b7a99c7148f8dab67296671.png
An exception which triggers a 503 HTTP status code

```
a:4:{i:0;s:25:"Cannot initialize product";i:1;s:9760:"#0 /data/web/magento2/var/generation/Magento/Review/Controller/Product/ListAjax/Interceptor.php(24): Magento\Review\Controller\Product\ListAjax->execute()
#1 /data/web/magento2/vendor/magento/framework/App/Action/Action.php(102): Magento\Review\Controller\Product\ListAjax\Interceptor->execute()
#2 /data/web/magento2/vendor/magento/module-review/Controller/Product.php(169): Magento\Framework\App\Action\Action->dispatch(Object(Magento\Framework\App\Request\Http))
#3 /data/web/magento2/vendor/magento/framework/Interception/Interceptor.php(74): Magento\Review\Controller\Product->dispatch(Object(Magento\Framework\App\Request\Http))
#4 /data/web/magento2/vendor/magento/framework/Interception/Chain/Chain.php(70): Magento\Review\Controller\Product\ListAjax\Interceptor->___callParent('dispatch', Array)
#5 /data/web/magento2/vendor/magento/framework/Interception/Chain/Chain.php(63): Magento\Framework\Interception\Chain\Chain->invokeNext('Magento\\Review\\...', 'dispatch', Object(Magento\Review\Controller\Product\ListAjax\Interceptor), Array, 'contextPlugin')
#6 /data/web/magento2/vendor/magento/module-store/App/Action/Plugin/Context.php(106): Magento\Framework\Interception\Chain\Chain->Magento\Framework\Interception\Chain\{closure}(Object(Magento\Framework\App\Request\Http))
#7 /data/web/magento2/vendor/magento/framework/Interception/Chain/Chain.php(67): Magento\Store\App\Action\Plugin\Context->aroundDispatch(Object(Magento\Review\Controller\Product\ListAjax\Interceptor), Object(Closure), Object(Magento\Framework\App\Request\Http))
#8 /data/web/magento2/vendor/magento/framework/Interception/Chain/Chain.php(63): Magento\Framework\Interception\Chain\Chain->invokeNext('Magento\\Review\\...', 'dispatch', Object(Magento\Review\Controller\Product\ListAjax\Interceptor), Array, 'customer-segmen...')
#9 /data/web/magento2/vendor/magento/module-customer-segment/Model/App/Action/ContextPlugin.php(81): Magento\Framework\Interception\Chain\Chain->Magento\Framework\Interception\Chain\{closure}(Object(Magento\Framework\App\Request\Http))
#10 /data/web/magento2/vendor/magento/framework/Interception/Chain/Chain.php(67): Magento\CustomerSegment\Model\App\Action\ContextPlugin->aroundDispatch(Object(Magento\Review\Controller\Product\ListAjax\Interceptor), Object(Closure), Object(Magento\Framework\App\Request\Http))
#11 /data/web/magento2/vendor/magento/framework/Interception/Chain/Chain.php(63): Magento\Framework\Interception\Chain\Chain->invokeNext('Magento\\Review\\...', 'dispatch', Object(Magento\Review\Controller\Product\ListAjax\Interceptor), Array, 'customer-app-ac...')
#12 /data/web/magento2/vendor/magento/module-customer/Model/App/Action/ContextPlugin.php(61): Magento\Framework\Interception\Chain\Chain->Magento\Framework\Interception\Chain\{closure}(Object(Magento\Framework\App\Request\Http))
#13 /data/web/magento2/vendor/magento/framework/Interception/Chain/Chain.php(67): Magento\Customer\Model\App\Action\ContextPlugin->aroundDispatch(Object(Magento\Review\Controller\Product\ListAjax\Interceptor), Object(Closure), Object(Magento\Framework\App\Request\Http))
#14 /data/web/magento2/vendor/magento/framework/Interception/Chain/Chain.php(63): Magento\Framework\Interception\Chain\Chain->invokeNext('Magento\\Review\\...', 'dispatch', Object(Magento\Review\Controller\Product\ListAjax\Interceptor), Array, 'storeCheck')
#15 /data/web/magento2/vendor/magento/module-store/App/Action/Plugin/StoreCheck.php(44): Magento\Framework\Interception\Chain\Chain->Magento\Framework\Interception\Chain\{closure}(Object(Magento\Framework\App\Request\Http))
#16 /data/web/magento2/vendor/magento/framework/Interception/Chain/Chain.php(67): Magento\Store\App\Action\Plugin\StoreCheck->aroundDispatch(Object(Magento\Review\Controller\Product\ListAjax\Interceptor), Object(Closure), Object(Magento\Framework\App\Request\Http))
#17 /data/web/magento2/vendor/magento/framework/Interception/Chain/Chain.php(63): Magento\Framework\Interception\Chain\Chain->invokeNext('Magento\\Review\\...', 'dispatch', Object(Magento\Review\Controller\Product\ListAjax\Interceptor), Array, 'weee-app-action...')
#18 /data/web/magento2/vendor/magento/module-weee/Model/App/Action/ContextPlugin.php(112): Magento\Framework\Interception\Chain\Chain->Magento\Framework\Interception\Chain\{closure}(Object(Magento\Framework\App\Request\Http))
#19 /data/web/magento2/vendor/magento/framework/Interception/Chain/Chain.php(67): Magento\Weee\Model\App\Action\ContextPlugin->aroundDispatch(Object(Magento\Review\Controller\Product\ListAjax\Interceptor), Object(Closure), Object(Magento\Framework\App\Request\Http))
#20 /data/web/magento2/vendor/magento/framework/Interception/Interceptor.php(138): Magento\Framework\Interception\Chain\Chain->invokeNext('Magento\\Review\\...', 'dispatch', Object(Magento\Review\Controller\Product\ListAjax\Interceptor), Array, 'tax-app-action-...')
#21 /data/web/magento2/vendor/magento/module-tax/Model/App/Action/ContextPlugin.php(91): Magento\Review\Controller\Product\ListAjax\Interceptor->Magento\Framework\Interception\{closure}(Object(Magento\Framework\App\Request\Http))
#22 /data/web/magento2/vendor/magento/framework/Interception/Interceptor.php(142): Magento\Tax\Model\App\Action\ContextPlugin->aroundDispatch(Object(Magento\Review\Controller\Product\ListAjax\Interceptor), Object(Closure), Object(Magento\Framework\App\Request\Http))
#23 /data/web/magento2/var/generation/Magento/Review/Controller/Product/ListAjax/Interceptor.php(39): Magento\Review\Controller\Product\ListAjax\Interceptor->___callPlugins('dispatch', Array, Array)
#24 /data/web/magento2/vendor/magento/framework/App/FrontController.php(55): Magento\Review\Controller\Product\ListAjax\Interceptor->dispatch(Object(Magento\Framework\App\Request\Http))
#25 /data/web/magento2/vendor/magento/framework/Interception/Interceptor.php(74): Magento\Framework\App\FrontController->dispatch(Object(Magento\Framework\App\Request\Http))
#26 /data/web/magento2/vendor/magento/framework/Interception/Chain/Chain.php(70): Magento\Framework\App\FrontController\Interceptor->___callParent('dispatch', Array)
#27 /data/web/magento2/vendor/magento/framework/Interception/Chain/Chain.php(63): Magento\Framework\Interception\Chain\Chain->invokeNext('Magento\\Framewo...', 'dispatch', Object(Magento\Framework\App\FrontController\Interceptor), Array, 'requestPreproce...')
#28 /data/web/magento2/vendor/magento/module-store/App/FrontController/Plugin/RequestPreprocessor.php(94): Magento\Framework\Interception\Chain\Chain->Magento\Framework\Interception\Chain\{closure}(Object(Magento\Framework\App\Request\Http))
#29 /data/web/magento2/vendor/magento/framework/Interception/Chain/Chain.php(67): Magento\Store\App\FrontController\Plugin\RequestPreprocessor->aroundDispatch(Object(Magento\Framework\App\FrontController\Interceptor), Object(Closure), Object(Magento\Framework\App\Request\Http))
#30 /data/web/magento2/vendor/magento/framework/Interception/Chain/Chain.php(63): Magento\Framework\Interception\Chain\Chain->invokeNext('Magento\\Framewo...', 'dispatch', Object(Magento\Framework\App\FrontController\Interceptor), Array, 'install')
#31 /data/web/magento2/vendor/magento/framework/Module/Plugin/DbStatusValidator.php(69): Magento\Framework\Interception\Chain\Chain->Magento\Framework\Interception\Chain\{closure}(Object(Magento\Framework\App\Request\Http))
#32 /data/web/magento2/vendor/magento/framework/Interception/Chain/Chain.php(67): Magento\Framework\Module\Plugin\DbStatusValidator->aroundDispatch(Object(Magento\Framework\App\FrontController\Interceptor), Object(Closure), Object(Magento\Framework\App\Request\Http))
#33 /data/web/magento2/vendor/magento/framework/Interception/Chain/Chain.php(63): Magento\Framework\Interception\Chain\Chain->invokeNext('Magento\\Framewo...', 'dispatch', Object(Magento\Framework\App\FrontController\Interceptor), Array, 'front-controlle...')
#34 /data/web/magento2/vendor/magento/module-page-cache/Model/App/FrontController/VarnishPlugin.php(55): Magento\Framework\Interception\Chain\Chain->Magento\Framework\Interception\Chain\{closure}(Object(Magento\Framework\App\Request\Http))
#35 /data/web/magento2/vendor/magento/framework/Interception/Chain/Chain.php(67): Magento\PageCache\Model\App\FrontController\VarnishPlugin->aroundDispatch(Object(Magento\Framework\App\FrontController\Interceptor), Object(Closure), Object(Magento\Framework\App\Request\Http))
#36 /data/web/magento2/vendor/magento/framework/Interception/Interceptor.php(138): Magento\Framework\Interception\Chain\Chain->invokeNext('Magento\\Framewo...', 'dispatch', Object(Magento\Framework\App\FrontController\Interceptor), Array, 'front-controlle...')
#37 /data/web/magento2/vendor/magento/module-page-cache/Model/App/FrontController/BuiltinPlugin.php(68): Magento\Framework\App\FrontController\Interceptor->Magento\Framework\Interception\{closure}(Object(Magento\Framework\App\Request\Http))
#38 /data/web/magento2/vendor/magento/framework/Interception/Interceptor.php(142): Magento\PageCache\Model\App\FrontController\BuiltinPlugin->aroundDispatch(Object(Magento\Framework\App\FrontController\Interceptor), Object(Closure), Object(Magento\Framework\App\Request\Http))
#39 /data/web/magento2/var/generation/Magento/Framework/App/FrontController/Interceptor.php(26): Magento\Framework\App\FrontController\Interceptor->___callPlugins('dispatch', Array, Array)
#40 /data/web/magento2/vendor/magento/framework/App/Http.php(135): Magento\Framework\App\FrontController\Interceptor->dispatch(Object(Magento\Framework\App\Request\Http))
#41 /data/web/magento2/vendor/magento/framework/App/Bootstrap.php(258): Magento\Framework\App\Http->launch()
#42 /data/web/magento2/pub/index.php(37): Magento\Framework\App\Bootstrap->run(Object(Magento\Framework\App\Http))
#43 {main}";s:3:"url";s:32:"/review/product/listAjax/id/3019";s:11:"script_name";s:10:"/index.php";}a
```

Normally a customer won't see this, but a search on Google says otherwise:
https://www.google.nl/search?q=503+product/listAjax/id

**Comment by magento-engcom-team** [comment]
@ruthger92, thank you for your report.
We've acknowledged the issue and added to our backlog.

**Comment by ananth-iyer** [comment]
I'm working on it #dmcdindia

**Comment by magento-engcom-team** [comment]
@Ananth747 thank you for joining. Please accept team invitation [here](https://github.com/orgs/magento/invitation) and self-assign the issue.

**Comment by ananth-iyer** [comment]
@magento-engcom-team When my pull request will be merged?

**Comment by ishakhsuvarov** [comment]
@Ananth747 Unfortunately, there is no exact estimate.

**Comment by magento-engcom-team** [comment]
Hi @ruthger92. Thank you for your report. 
The issue has been fixed in magento/magento2#15369 by @Ananth747 in 2.2-develop branch
Related commit(s):
 - [a1d0e3b2b421881aecb288f91b0399a63796ec73](https://github.com/magento/magento2/commit/a1d0e3b2b421881aecb288f91b0399a63796ec73)
 - [b81bf6751f8fd736087d309136eb77b57a1a35e1](https://github.com/magento/magento2/commit/b81bf6751f8fd736087d309136eb77b57a1a35e1)
 - [b0848c452ceae57c6af898665fd68d7d88ba98a8](https://github.com/magento/magento2/commit/b0848c452ceae57c6af898665fd68d7d88ba98a8)
 - [f5bd1c688af81337048eefcb990a9900dfec78d2](https://github.com/magento/magento2/commit/f5bd1c688af81337048eefcb990a9900dfec78d2)

The fix will be available with the upcoming 2.2.7 release.

**Comment by sidolov** [comment]
Hi @ruthger92. Thank you for your report. 
The issue has been fixed in magento/magento2#17632 by @mage2pratik in 2.1-develop branch
Related commit(s):
 - [81c0320faf3fc5df6dee1227d22651604ade8b57](https://github.com/magento/magento2/commit/81c0320faf3fc5df6dee1227d22651604ade8b57)
 - [c729a3bb2bec3a505c4ab523ad04897b0519be2c](https://github.com/magento/magento2/commit/c729a3bb2bec3a505c4ab523ad04897b0519be2c)
 - [53ba0ac25195a7474971e6ab03257386a9113977](https://github.com/magento/magento2/commit/53ba0ac25195a7474971e6ab03257386a9113977)
 - [01a7e20deea3e72a27b8784278eb5eb5ba6bc1fb](https://github.com/magento/magento2/commit/01a7e20deea3e72a27b8784278eb5eb5ba6bc1fb)

The fix will be available with the upcoming 2.1.16 release.

**Comment by magento-engcom-team** [comment]
Hi @ruthger92. Thank you for your report. 
The issue has been fixed in magento/magento2#17721 by @nmalevanec in 2.3-develop branch
Related commit(s):
 - [3f0168cbf6589e850cbc4ca4b6ca3796a04330ab](https://github.com/magento/magento2/commit/3f0168cbf6589e850cbc4ca4b6ca3796a04330ab)

The fix will be available with the upcoming 2.3.0 release.
